### PR TITLE
Add integer parsing with unit support

### PR DESCRIPTION
**Pull request by PragmaTwice**
- Add `ParseSizeAndUnit`
- Clean some code

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]

File: src/common/parse_util.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ */
+
+#include "parse_util.h"
+
+#include <limits>
+
+// num << bit <= MAX  ->  num <= MAX >> bit
+template <typename T, typename U>
+StatusOr<T> CheckedShl(T num, U bit) {
```
Comment (torwig) [inline src/common/parse_util.cc]: Is it better to name it as `CheckedShiftL` (to describe the main operation which is a **shift** and stress on the letter `L`) or even `CheckedShiftLeft`?

Comment (PragmaTwice) [reply src/common/parse_util.cc]: done

**Review by torwig — APPROVED** [review]

**Comment by tisonkun** [comment]
@PragmaTwice Could you please link the motivation of adding such a feature?

**Comment by PragmaTwice** [comment]
> @PragmaTwice Could you please link the motivation of adding such a feature?

It is needed in #953, and the implementation in #953 is not good to me so I do not want to refactor these code after merge it.

**Review by tisonkun — APPROVED** [review]
LGTM. Merging...

**Pull request merged**